.github/workflows/unit_tests.yaml

```diff
@@ -0,0 +1,33 @@
+name: Execute the project's unit tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10']
+
+    name: 'Python ${{matrix.python-version}} test'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip poetry
```

Comment:
Why do you need to upgrade pip?

Reply:
I added a matrix with a single value so it could be easily updated when future Python versions were released.
As for the comment about upgrading pip, now that you say it I think I shouldn't have done it. I'll fix that before merging.